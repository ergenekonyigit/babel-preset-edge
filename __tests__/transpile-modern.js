import buildPreset from "../src"
import { fixtures, titles, check } from "./core"

describe("Transpile: Modern", () => {
  const options = buildPreset(null, {
    sourceMaps: false,
    transpile: "modern"
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
