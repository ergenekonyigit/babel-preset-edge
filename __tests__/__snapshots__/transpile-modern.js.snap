// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpile: Modern comments 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

function translate(value) {
  return _camelCase(value);
}"
`;

exports[`Transpile: Modern deadcode 1`] = `
"const STATIC = false;
before();
after();"
`;

exports[`Transpile: Modern es2015: classes 1`] = `
"export class Component extends Other {
  constructor() {
    super();
    this.flag = true;
  }

  main() {
    alert(this.foo);
  }

}"
`;

exports[`Transpile: Modern es2015: for_off 1`] = `
"function test() {
  for (const taskName of tasks) {
    try {
      executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Transpile: Modern es2015: generator 1`] = `
"function* idMaker() {
  var index = 0;

  while (true) yield index++;
}"
`;

exports[`Transpile: Modern es2015: set 1`] = `"new Set([1, 2, 3]);"`;

exports[`Transpile: Modern es2016: exponentiation 1`] = `"var result = 8;"`;

exports[`Transpile: Modern esnext: async 1`] = `
"async function wait() {
  await loadData();
}"
`;

exports[`Transpile: Modern esnext: async_loop 1`] = `
"async function executeTasks(tasks) {
  for (const taskName of tasks) {
    try {
      await executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Transpile: Modern esnext: class properties 1`] = `
"import _defineProperty from \\"@babel/runtime/helpers/builtin/es6/defineProperty\\";
export class Component {
  constructor() {
    _defineProperty(this, \\"onClick\\", () => {});
  }

}"
`;

exports[`Transpile: Modern esnext: dynamic import 1`] = `
"import(\\"lodash\\").then(lodash => {
  console.log(lodash.camelCase(\\"hello world\\"));
});"
`;

exports[`Transpile: Modern esnext: object spread 1`] = `
"import _objectSpread from \\"@babel/runtime/helpers/builtin/es6/objectSpread\\";

let original = {
  bar: 10
},
    variant = _objectSpread({
  foo: 1
}, original);"
`;

exports[`Transpile: Modern loadable 1`] = `
"import loadable from 'loadable-components';
export const Home = loadable(() => import(\\"./Home\\"), {
  modules: [\\"./Home\\"]
});
export const About = loadable(() => import(\\"./About\\"), {
  modules: [\\"./About\\"]
});
export const Contact = loadable(() => import(\\"./Contact\\"), {
  modules: [\\"./Contact\\"]
});"
`;

exports[`Transpile: Modern lodash: async 1`] = `
"import _parallel from \\"async/parallel\\";

_parallel([]);"
`;

exports[`Transpile: Modern lodash: lodash 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

_camelCase(\\"hello world\\");"
`;

exports[`Transpile: Modern lodash: recompose 1`] = `
"import _compose from \\"recompose/compose\\";

_compose([]);"
`;

exports[`Transpile: Modern react: jsx 1`] = `
"function render() {
  return React.createElement(\\"h1\\", null, \\"Hello\\");
}"
`;

exports[`Transpile: Modern react: jsx spread 1`] = `
"function render(props) {
  return React.createElement(\\"h1\\", Object.assign({
    className: \\"large\\"
  }, props), \\"Hello\\");
}"
`;

exports[`Transpile: Modern react: proptypes 1`] = `
"import React from \\"react\\";
import PropTypes from \\"prop-types\\";

class MyComponent extends React.Component {
  render() {
    return React.createElement(\\"div\\", null, this.props.userName);
  }

}

MyComponent.propTypes = {
  userName: PropTypes.string
};"
`;
